Happy Dance/Assets/_Scripts/GameTimer.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System.Collections;
+
+using UnityEngine;
+using UnityEngine.Assertions;
+using UnityEngine.Playables;
+
+using Liminal.Core.Fader;
+using Liminal.SDK.Core;
+
+/// <summary>
+/// GameTimer is sued to control the ending of the happy dance experience.
+/// </summary>
+public class GameTimer 
+    : MonoBehaviour
+{
+    public float SecondsLeft;
+    public float VolumeFadeSpeed;
+    public DanceDetector PlayerDanceDetector;
+    public PlayableDirector Director;
+
+    void Start()
+    {
+        Assert.IsNotNull(PlayerDanceDetector, "PlayerDanceDetector must not be null");
+        StartCoroutine(TimerCoro());
+    }
+
+	IEnumerator TimerCoro()
+	{
+        while (SecondsLeft > 0)
+        {
+            SecondsLeft -= Time.deltaTime;
+            yield return new WaitForEndOfFrame();
+        }
+
+        PlayerDanceDetector.Active = false;
+
+        yield return new WaitForSeconds(5f);
+        Director.gameObject.SetActive(true);
+
+        yield return new WaitForSeconds((float) Director.duration + 0.1f);
```

Comment:
director has an actual state you can read. So you can actually mess around with it in the timeline and this will still wait for it.

is there a particular reason we're just waiting based on duration?

yield return new WaitUntil(() => !Director.IsPlaying)